core/src/autogluon/core/hpo/resources_calculator.py

```diff
@@ -0,0 +1,172 @@
+import psutil
+
+from abc import ABC, abstractmethod
+from ray_lightning.tune import get_tune_resources
+
+
+class ResourceCalculator(ABC):
+    
+    @property
+    @abstractmethod
+    def type(self):
+        """Type of the resource calculator"""
+        return NotImplementedError
+    
+    @abstractmethod
+    def get_resources_per_trial(self, **kwargs) -> dict:
+        """Calculate resources per trial and return additional info"""
+        raise NotImplementedError
+    
+
+class CpuResourceCalculator(ResourceCalculator):
+    
+    @property
+    def type(self):
+        return 'cpu'
```

Comment:
nit: potentially a bad name for property. `type` is a [built-in function](https://docs.python.org/3/library/stdtypes.html#type-objects).